### ISSUE TITLE
Test results are not sending

### ISSUE THREAD

**Issue by Khrove** [carmenmitru/wdio-slack-service#1]
Node version: 10.16.2
NPM version: 6.13.7
WDIO version: 5.18.4
wdio-slack-service version: 2.0.7

Issue that is occuring:
The only message that is sent to Slack is "Running Tests Report" when *I think* it's expected that I should receive "Running Tests Report" and then some message containing the test result attachments.

Config:
`
exports.config = {
 updateJob: false,
    maxInstances: 1,
user: bstack_user,
    key: bstack_key,
    bail: 2,
    capabilities: [{
        'os': browserObj.os,
        'os_version': browserObj.os_version,
        'browserName': process.env.BROWSER,
        'browser_version': browserObj.browser_version,
        build: browserConfig.BUILD,
        'project': browserConfig.PROJECT,
        "sessionName": global.currentTestName,
        'browserstack.local': true,
        'browserstack.debug': true,
        acceptInsecureCerts: true,
        acceptSslCert: true,
        resolution: browserConfig.RESOLUTION
    }],
    reporters: ['spec',
        [HtmlReporter, {
            outputDir: './reports/html-reports',
            filename: 'reports.html',
            reportTitle: 'Test Reports',
            showInBrowser: false,
            useOnAfterCommandForScreenshot: true
        }],
        ['timeline', {
            outputDir: './timeline_reports',
            embedImages: true,
            images: {
                quality: 80,
                resize: false,
                reductionRatio: 2
            },
            screenshotStrategy: 'none'
        }]
    ],
    services: [
        [slack, {
            webhook: process.env.SLACK_WEBHOOK_URL || "...",
        }],
        [TimelineService]
    ],
logLevel: 'warn',
    coloredLogs: true,
    baseUrl: partnerURL,
    waitforTimeout: 20000,
    host: 'hub.browserstack.com',
    framework: 'mocha',
    mochaOpts: {
        ui: 'bdd',
        timeout: '65000',
        compilers: ['@babel/register']
    },
// ... lifecycle hooks
}
`

**Comment by tomasfly** [comment]
Have you found any solution on this? I am getting the same. Just "Running Tests Report

"

**Comment by Khrove** [comment]
No, I have not found a solution to this yet. Someone will probably have to fork the repo and debug it because it doesn't look like this is being actively maintained.

**Comment by tomasfly** [comment]
I realized it seems to notify correctly to the Slack channel when you use in framework: 'jasmine' or framework: 'mocha' in testrunner configuration. But it doesnt seem to notify when you use 'cucumber'. Which framework are you using?

**Comment by aetjansen** [comment]
I too get only "**Running Tests Report**" and nothing else in the slack channel. So, it posts to Slack correctly, but not the correct info. I tried with both mocha and jasmine as frameworks used. Using wdio v6.1.11 and current npm package for this slack service.

**Comment by adrian-brown** [comment]
Great idea, any chance this is still supported @carmenmitru ?  And positioned to work across Jasmine, Mocha and Cucumber frameworks?

**Comment by soumyaevan** [comment]
I am also facing the same issue. Any update yet?

**Comment by carmenmitru** [comment]
Hello @soumyaevan, @adrian-brown , @aetjansen @aetjansen , @tomasfly , @Khrove 

I updated the npm package. Let me know if now works. I also added the notifyOnlyOnFailure option

**Comment by himanshujane3** [comment]
Hello @carmenmitru  
I'm using 
"wdio-slack-service": "^2.0.8" and notifyOnlyOnFailure: true, 

**Observation**
Notification is sent even when all the test are passed 
And also getting this error
Error in "AfterTest Hook"
Cannot read property 'matcherResult' of undefined

When notifyOnlyOnFailure is set to false - It doesnt send the notification at all.. and i still get the AfterTest Hook error

It seems like flag is set vice versa and there is a bug as well. 
It will be great if someone can fix this.. Thanks in advance. 

![image](https://user-images.githubusercontent.com/37361620/86433906-903d7400-bd19-11ea-8488-2463e04ee20f.png)

**Comment by mcf1727** [comment]
If I set notifyOnlyOnFailure to true, I can receive reports. But no if it's set as false

**Comment by mcf1727** [comment]
with Mocha, even the test failed, in the slack report it's counted as a passed test.   I don't know why

**Comment by AutomationReddy** [comment]
This is fixed in latest version

**Comment by xinqilin** [comment]
Hi !
Sorry, I have some problem as same as this issue

my dependencies: 
```
"dependencies": {
    "@wdio/cli": "^7.9.0",
    "request": "^2.88.2",
    "request-promise": "^4.2.6",
    "wdio-wiremock-service": "^6.0.6"
  },
  "devDependencies": {
    "@wdio/firefox-profile-service": "^7.8.0",
    "@wdio/local-runner": "^7.8.0",
    "@wdio/mocha-framework": "^7.8.0",
    "@wdio/selenium-standalone-service": "^7.8.0",
    "@wdio/spec-reporter": "^7.8.0",
    "chromedriver": "^91.0.1",
    "wdio-chromedriver-service": "^7.2.0",
    "wdio-slack-service": "^2.0.9"
  },
```

in my ``wdio.conf.js`` setting
```
            slack, {
                .....
                notifyOnlyOnFailure: true,  // Send notification only on test failure
               .......
            }
``` 

and my test report in console 
``Spec Files:      0 passed, 1 failed, 1 total (100% completed) in 00:00:09``

and there is nothing events push to my slack channel,

Is there any tips to help me, please   T_T

**Comment by xinqilin** [comment]
Sorry  ,  I got where the problem is
in the ``index.js`` file
``afterTest()`` method
the line 
```
let testError = results.error.matcherResult.message().replace(/[\u001b\u009b][-[+()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g, "");

```
result.error.matcherResult  always undefined  ,   so ``.message()``  is not fuctioning

**Comment by DiMavrodiev** [comment]
Hi everyone, 

I have been having the same problem as the people above. I have `"wdio-slack-service": "^2.0.9"` and I use `mocha` for test running. I get all PASS tests in my slack channel but none of the failing tests are getting into to channel. Any ideas how can I see the failing tests in slack. `notifyOnlyOnFailure: true` doesn't help
